include/platform/mir/graphics/platform.h

```diff
@@ -118,6 +119,38 @@ enum PlatformPriority : uint32_t
                          */
 };
 
+struct SupportedDevice
+{
+    /**
+     * UDev device the platform claims
+     *
+     * Mir assumes that all devices that platforms can claim are represented in udev,
+     * and can be uniquely identified there. Particularly: platform loading will guarantee
+     * that at most one platform (of each type) is created on each unique device.
+     *
+     * \note    It is acceptable for device to be empty; in this case Mir assumes that
+     *          device is incomparable with any other device. In particular, hosted platforms
+     *          like X11 or Wayland should use an empty device, as they are not bound to any
+     *          particular hardware.
+     */
+    std::unique_ptr<udev::Device> device;
+    PlatformPriority support_level;     /**< How well the platform can support this device */
+
+    class DriverPrivate
+    {
+    public:
+        DriverPrivate() = default;
+        virtual ~DriverPrivate() = default;
+    };
+    /**
+     * Platform-private data from probing
+     *
+     * If there is any extra data helpful in creating a Platform on this device, the platform can
+     * stash it here, and it will be passed in to the platform constructor.
+     */
+    std::unique_ptr<DriverPrivate> platform_data;
+};
```

Comment:
Maybe this would be better as an interface than as `struct`.

Movable, non-copyable structs are certainly legal, but that means they are also mutable and I'm not sure that's ideal. (I wondered at first why this wasn't immutable)

Reply:
Yeah, it's annoying that *movable* and *immutable* are mutually exclusive.

I don't have a particularly strong opinion here (although it's mildly convenient elsewhere in the code that this is *not* immutable).